### DISCUSSION TITLE
question - how to generate those .list files

### DISCUSSION THREAD

**Discussion by cjbd** [Eloston/ungoogled-chromium#1391] [Q&A] [answered]
hello, this is really an awesome project, your approach to ungoogle chrome is amazing!

btw, where're those '.list' files from, especially the `domain_substitution.list` file? is there some script to scan the entire chromium source code to generate that? is it possible to open source that script too?

thank you!

**Comment by wchen342** [comment, marked as answer]
See [devutils/update_lists.py](https://github.com/Eloston/ungoogled-chromium/blob/master/devutils/update_lists.py).

**Comment by cjbd** [comment]
@wchen342 , thanks a lot!

**Comment by cjbd** [comment]
sorry, i have some follow up questions:

1. what about `domain_regex.list`, seems there's no script to generate this one

2. can this project be used for chromiumos source to substitute google domains?

thank you!

**Comment by wchen342** [comment]
1. I think the list is composed by adding known Google domains manually. Is that correct? @Eloston 
2. Never tried that but if the source tree is the same then yes it is possible.

**Comment by Eloston** [comment]
> I think the list is composed by adding known Google domains manually. Is that correct? @Eloston

Yes

@cjbd You may also find the design docs in [`docs/design.md`](https://github.com/Eloston/ungoogled-chromium/blob/master/docs/design.md) helpful.
